src/ProviderBasedGenerator.php

```diff
@@ -49,6 +49,8 @@ public function generate($name, $parameters = [], $referenceType = self::ABSOLUT
     {
         if ($name instanceof SymfonyRoute) {
             $route = $name;
+        } elseif ('' === $name && array_key_exists(RouteObjectInterface::ROUTE_OBJECT, $parameters) && $parameters[RouteObjectInterface::ROUTE_OBJECT] instanceof SymfonyRoute) {
```

Comment:
i think that using RouteObjectInterface::ROUTE_OBJECT as key is fine here. we would use the new magic route name to trigger this, instead of `''`. please define as a constant in the RouteObjectInterface.

Reply:
So using `cmf_routing_object` instead of an empty string?

Reply:
Yes this is an idea. Just create new constant in `RouteObjectInterface` i.e. `RouteObjectInterface::OBJECT_BASED_ROUTE_NAME` and use it instead of an empty string.

Reply:
aight 👍